apps/wallet/src/ui/app/pages/qredo-connect/SelectQredoAccountsPage.tsx

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ArrowRight16 } from '@mysten/icons';
+import { useEffect, useState } from 'react';
+import { toast } from 'react-hot-toast';
+import {
+    useParams,
+    useLocation,
+    Navigate,
+    useNavigate,
+} from 'react-router-dom';
+
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { Button } from '../../shared/ButtonUI';
+import { SelectQredoAccountsSummaryCard } from './components/SelectQredoAccountsSummaryCard';
+import { useQredoUIPendingRequest } from './hooks';
+import { PasswordInputDialog } from '_components/menu/content/PasswordInputDialog';
+import Overlay from '_components/overlay';
+import { type Wallet } from '_src/shared/qredo-api';
+
+export function SelectQredoAccountsPage() {
+    const { id } = useParams();
+    const { state } = useLocation();
+    const navigate = useNavigate();
+    const backgroundService = useBackgroundClient();
+    const qredoRequestReviewed = !!state?.reviewed;
+    const { data: qredoPendingRequest, isLoading: isQredoRequestLoading } =
+        useQredoUIPendingRequest(id);
+    // do not call the api if user has not clicked continue in Qredo Connect Info page
+    const fetchAccountsEnabled =
+        !isQredoRequestLoading &&
+        (!qredoPendingRequest || qredoRequestReviewed);
+
+    const [selectedAccounts, setSelectedAccounts] = useState<Wallet[]>([]);
+    const [showPassword, setShowPassword] = useState(false);
+    const shouldCloseWindow =
+        (!isQredoRequestLoading && !qredoPendingRequest) || !id;
+    useEffect(() => {
+        if (shouldCloseWindow) {
+            window.close();
```

Comment:
should the whole window close or should we just redirect users to the home page / previous step?

Reply:
Yeah should be fine we do the same for transactions and connections